tools/cryptogenotyper/cryptogenotyper.xml

```diff
@@ -0,0 +1,141 @@
+<tool id="CryptoGenotyper" name = "CryptoGenotyper" version="@VERSION@">
+  <description>
+    CryptoGenotyper is a standalone tool to analyze Cryptosporidium Sanger sequencing data and classify the species and subtype based on SSU rRNA and gp60 gene markers, respectively.
+  </description>
+  <macros>
+   <token name="@VERSION@">1.0</token>
+  </macros>
+  <requirements>
+    <requirement type="package" version ="@VERSION@">cryptogenotyper</requirement>
+  </requirements>
+  <version_command>cryptogenotyper --version</version_command>
+  <command detect_errors="exit_code">
+     <![CDATA[
+
+     #set $ref_file=''
+
+     #if $reference.ref == "no"
+        ln -s "${reference.db}" "${reference.db.name}" &&
+        #set $ref_file = $reference.db.name
+     #end if
+
+     #if $primers["seqtype"] == "contig"
+        ln -s "${$primers.abi_input["forward"]}" "${primers.abi_input.name}_forward.ab1" &&
+        ln -s "${$primers.abi_input["reverse"]}" "${primers.abi_input.name}_reverse.ab1" &&
+        cryptogenotyper -i "." -m "$marker" -t "$primers.seqtype" -f "forward" -r "reverse"
+        #if $reference.ref == "no"
+          --databasefile $ref_file
+        #end if
+     #else
+        ln -s "${primers.abi_input}" "${primers.abi_input.element_identifier}" &&
+        cryptogenotyper -i "./${primers.abi_input.element_identifier}" -m "$marker" -t "$primers.seqtype" -f "" -r ""
```

Comment:
what does -f and -r is doing?

Reply:
In this case these parameters are redundant. They are used to filter input files based on names of the primers, but since in Galaxy will work with maximum a pair of files at a time, then it is redundant indeed. I had removed it